### PR TITLE
Meta+Documentation: Require Xcode 14.3 or Clang 14 for the host compiler

### PR DESCRIPTION
**Pull request by BertalanD**
There have been multiple reports of Xcode 14.0 (based on upstream LLVM 14) segfaulting when compiling `LibCore/Process.cpp`. Let's require Xcode 14.3, which is a known good version based on LLVM 15.

Note that Xcode 14.3 requires macOS Ventura, so users of Monterey or older are expected to get Homebrew Clang instead.

Homebrew Clang 13 also suffers from the same crash. Although I have not tested on Linux, the backtrace points to the middle-end, so x86_64 is also likely to be affected. LLVM 14 was released 14 months ago, so it's not an unreasonable requirement.

### CODE REVIEW

File: Meta/serenity.sh [hunk 1]
```diff
@@ -203,7 +205,11 @@ pick_host_compiler() {
         return
     fi
 
-    die "Please make sure that GCC version 12, Clang version 13, or higher is installed."
+    if [ "$(uname -s)" = "Darwin" ]; then
+        die "Please make sure that Xcode 14.3, Homebrew Clang 13, or higher is installed."
```
Comment (ADKaster) [inline Meta/serenity.sh:209]: Homebrew clang 13 is unlikely to work :)

Comment (BertalanD) [reply Meta/serenity.sh:209]: Ah, let me test it and update the search list, then

Comment (BertalanD) [reply Meta/serenity.sh]: Updated the requirements to LLVM Clang 14.

File: Meta/serenity.sh [hunk 2]
```diff
@@ -191,14 +191,14 @@ pick_host_compiler() {
...
-    find_newest_compiler clang clang-14 clang-15 /opt/homebrew/opt/llvm/bin/clang
+    find_newest_compiler clang clang-14 clang-15 clang-16 /opt/homebrew/opt/llvm/bin/clang
     if is_supported_compiler "$HOST_COMPILER"; then
         export CC="${HOST_COMPILER}"
         export CXX="${HOST_COMPILER/clang/clang++}"
         return
     fi
 
-    find_newest_compiler egcc gcc gcc-12 /usr/local/bin/gcc-12 /opt/homebrew/bin/gcc-12
+    find_newest_compiler egcc gcc gcc-12 gcc-13 /usr/local/bin/gcc-{12,13} /opt/homebrew/bin/gcc-{12,13}
```
Comment (ADKaster) [inline Meta/serenity.sh:201]: Does this kind of expansion work in the ancient bash macOS ships by default?

Comment (BertalanD) [reply Meta/serenity.sh:201]: It definitely does work under Homebrew bash, which is a build dependency.

I have just tested `/bin/bash` (version 3.2) now, and it works there as well.

**Review by GMTA — APPROVED** [review]

**Pull request merged**